### ISSUE TITLE
Response type mappings

### ISSUE THREAD

**Issue by sebdehne** [springfox/springfox#501]
Hi,

We are evaluating swagger and have some questions. We are using swagger-springmvc version 0.9.1. 

Quesion: My handler methods use ResponseEntity<Object> because they respond with different types depending on the status code. And I am using @ApiResponses/@ApiResponse to specify which response type is returned @ which response code. However, the /api-docs json does not reflect this. For this annotations:

```
@ApiResponses({
        @ApiResponse(code = 404, message = "Not Found", response = Void.class),
        @ApiResponse(code = 200, message = "OK", response = OrderContent.class),
})
```

I get:

{
  code: 200,
  message: "OK",
  responseModel: "ResponseEntity«object»"
},
{
code: 404,
message: "Not Found",
responseModel: "Void"
}

How can I make "responseModel" return OrderContent.class?

Basically, what I want is that DefaultResponseMessageReader.applyReturnTypeOverride() is skipped entirely because the return type of the method is no indication to swagger which type is actually returned for 200 OK. For that, I want to use the @ApiResponse annotation and want swagger to ignore the return type of the method.

Thanks,
Sebastian
